frontend/public/components/utils/units.js

```diff
@@ -58,6 +58,56 @@ const getType = (name) => {
   return type;
 };
 
+const log = (x, y) => {
+  return Math.log(y) / Math.log(x);
+};
+
+//Find the unit which is shared by most of the data units
+const consensusLevel = (bestLevel) => {
+  const levels = {};
+  bestLevel.forEach((e) => {
+    if (levels[e]) {
+      levels[e] += 1;
+    } else {
+      levels[e] = 1;
+    }
+  });
+  const mostUsed = Object.keys(levels).sort((x, y) => levels[y] - levels[x])[0];
+  return mostUsed;
+};
+
+//Find the best unit for the units
+const appropriateLevel = (dataPoints, type) => {
+  const base = type.divisor;
+  const y = [];
+  dataPoints.forEach((point) => {
+    //Remove trailing decimal points
+    y.push(Number(log(base, point.y)).toFixed(0));
+  });
+  //Which is the best level
+  const bestLevel = consensusLevel(y);
+  return bestLevel;
+};
```

Comment:
These two functions should probably be combined into one. Here's an example of how that could look (untested):

```js
//Find the unit of measurement that best represents a data set.
const mostFrequentUnit = (data, type) => {

  // Reduces the data set to an array containing the following elements in this order:
  //  - the most frequently occurning unit in the data set,
  //  - the number of occurences of that unit,
  //  - an object used to track counts for every unit which occurs in the data set.
  // This allows us to iterate once, and get all of the info we need to return the most frequently occuring unit.
  const [mostFrequentUnit] = data.reduce((accumulator, point) => {
    const [maxUnit, maxCount, counts] = accumulator; // use es6 destructuring to track counts and max simultaneously
    const index = Math.floor(log(_.get(type, divisor, 1000), point.y));
    const unit = _.get(type, ['units', om], '');
    const unitCount = _.get(counts, unit, 0) + 1;
    return [
      ...(unitCount > maxCount) ? [unit, unitCount] : [maxUnit, maxCount],
      {
        ...counts,
        [unit]: unitCount
      }
    ];
  }, ['', 0, {}]);
  return mostFrequentUnit;
};
```

Also, since this logic is pretty complex, some better naming and more detailed comments would be nice.

Reply:
If we end up creating a new prop on the Area/AreaChart component as I suggest in my comments, this will also need to be exported.